### DISCUSSION TITLE
dnscrypt-proxy appears to not be included in Debian bookworm repositories, is that intentional?

### DISCUSSION THREAD

**Discussion by f1sherman** [DNSCrypt/dnscrypt-proxy#2410] [Q&A] [answered]
I recently upgraded from Debian bullseye to bookworm, and `apt install dnscrypt-proxy` results in a `E: Unable to locate package dnscrypt-proxy` error. Was it intentionally removed?

**Comment by lifenjoiner** [comment]
What I can see from the [Debian Package Search Results](https://packages.debian.org/search?keywords=dnscrypt&searchon=names&suite=stable&section=all):
1. It is really out of date,
2. The maintainer seems to be from debian.org.

So, it is suggested to install it manually following [the generic instructions for Linux](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Installation-linux).

**Reply by f1sherman** [reply]
Ok sounds good! Would it be a good idea to update the [installation instructions](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Installation-on-Debian-and-Ubuntu#debian-testing-bullseye-debian-unstable-sid) with the latest guidance?

**Reply by jedisct1** [reply]
Yes it would! Please do!

**Reply by f1sherman** [reply]
I don't seem to have access to update the wiki, but would be happy to update it if you can give me access!

**Comment by jedisct1** [comment, marked as answer]
The reason is that a 3rd-party dependency (that isn't used by `dnscrypt-proxy` itself, except for testing) called `powerman` didn't compile any more on Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1017302 and it wasn't fixed before the freeze.

**Reply by f1sherman** [reply]
Good to know, thanks!

**Comment by a7md0** [comment]
There is a workaround to install dnscrypt-proxy on Debian 12 using apt from the `sid` unstable repository as described in [How to install dnscrypt-proxy on Debian Linux 11/12](https://www.cyberciti.biz/faq/installing-dnscrypt-proxy-on-debian-linux/). No need to build from source or download the binaries manually.

---

To install `dnscrypt-proxy` from Debian *unstable* (sid) without upgrading your entire system, follow these steps:

1. **Add the Debian sid repository:**

```bash
echo 'deb http://deb.debian.org/debian sid main' | sudo tee /etc/apt/sources.list.d/sid.list
```

2. **Set up APT pinning to prevent packages from sid being installed by default:**

```bash
sudo tee /etc/apt/preferences.d/99-sid-repository-pinning > /dev/null <<EOF
Package: *
Pin: release a=unstable
Pin-Priority: -10
EOF
```

3. **Update package lists and verify pinning:**

```bash
sudo apt update
sudo apt-cache policy
sudo apt-cache policy dnscrypt-proxy
```

4. **Install `dnscrypt-proxy` specifically from sid:**

```bash
sudo apt install -t sid dnscrypt-proxy
```

**Reply by Ratione** [reply]
Doing \<preferred-deity\>'s work. I was looking for a solution to this issue for a while.
